File: uimodule/src/main/java/com/lpjeremy/uimodule/view/LoadingStateView.java
package com.lpjeremy.uimodule.view;

import android.content.Context;
import androidx.annotation.Nullable;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.lpjeremy.uimodule.R;

/**
 * @desc:加载状态展示view
 * @date:2019/2/11 10:48
 * @auther:lp
 * @version:1.0
 */
public class LoadingStateView extends LinearLayout {
    ImageView mImageView;
    TextView mTextView;

    public LoadingStateView(Context context) {
        super(context);
    }

    public LoadingStateView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public LoadingStateView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        mImageView = findViewById(R.id.imageView);
        mTextView = findViewById(R.id.textView);
        hide();
    }

    /**
     * 加载成功
     */
    public void hide() {
        setVisibility(GONE);
    }

    /**
     * 加载失败，默认提示文本，无提示图片
     */
    public void onLoadFail() {
        onLoadFail(0, "加载失败！！！");
    }

    /**
     * 加载失败，设置提示文本，无提示图片
     * @param failToast
     */
    public void onLoadFail(String failToast) {
        onLoadFail(0, failToast);
    }

    /**
     * 加载失败，设置提示图片和文本
     *
     * @param resId     提示图片
     * @param failToast 提示文本
     */
    public void onLoadFail(int resId, String failToast) {
        setVisibility(VISIBLE);
        if (resId > 0) {
            mImageView.setVisibility(VISIBLE);
            mImageView.setImageResource(resId);
        } else {
            mImageView.setVisibility(GONE);
        }
        if (TextUtils.isEmpty(failToast)) {
            mTextView.setText("加载失败！！！");
        } else {
            mTextView.setText(failToast);
        }
    }
}

File: app/src/main/java/com/hysyyl/bloodapp/base/view/LoadingStateView.kt
package com.hysyyl.bloodapp.base.view

import android.content.Context
import android.os.Handler
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.layout_loading_state.view.*

class LoadingStateView(context: Context?, attrs: AttributeSet?) : LinearLayout(context, attrs) {
    private val mHandler: Handler = Handler()

    override fun onFinishInflate() {
        super.onFinishInflate()
        showLoading(null)
    }

    /**
     * -1 隐藏LoadingStateView
     * 0 显示loadingImage 动画
     * 1 显示loadingImage 动画 + 文字
     * 2 显示图片提示 + 文字
     *
     */
    private fun showLoadingStateView(state: Int, toast: String?, imgResId: Int?) {
        mHandler.post {
            loadingTxtToast.text = toast
            if (imgResId != null && imgResId > 0) {
                loadingImgToast.setImageResource(imgResId)
            }
            when (state) {
                -1 -> {
                    visibility = View.GONE
                    loadingImage.stopLoading()
                }
                0 -> {
                    visibility = View.VISIBLE
                    loadingTxtToast.visibility = View.GONE
                    loadingImgToast.visibility = View.GONE
                    loadingImage.visibility = View.VISIBLE
                    loadingImage.startLoading()

                }
                1 -> {
                    visibility = View.VISIBLE
                    loadingTxtToast.visibility = View.VISIBLE
                    loadingImgToast.visibility = View.GONE
                    loadingImage.visibility = View.VISIBLE
                    loadingImage.startLoading()
                }
                2 -> {
                    visibility = View.VISIBLE
                    loadingTxtToast.visibility = View.VISIBLE
                    loadingImgToast.visibility = View.VISIBLE
                    loadingImage.visibility = View.GONE
                    loadingImage.stopLoading()
                }
            }
        }
    }

    /**
     * 隐藏loading view
     */
    fun hideLoadingView() {
        showLoadingStateView(-1, "", 0)
    }

    /**
     * 显示loading动画
     * @param toast 传null只显示loading图片动画  传空字符  显示loading图片动画和‘努力加载中...’
     */
    fun showLoading(toast: String?) {
        if (toast == null) {
            showLoadingStateView(0, "", 0)
        } else {
            var msg = if (toast.isEmpty()) "努力加载中..." else toast
            showLoadingStateView(1, msg, 0)
        }

    }

    /**
     * 显示加载结果
     */
    fun showLoadResult(toast: String, imgResId: Int) {
        showLoadingStateView(2, toast, imgResId)
    }

}